scipy/special/sf_error.cc

```diff
@@ -54,10 +62,13 @@ void sf_error_v(const char *func_name, sf_error_t code, const char *fmt, va_list
         PyOS_snprintf(msg, 2048, "scipy.special/%s: %s", func_name, sf_error_messages[(int) code]);
     }
 
+#ifndef Py_GIL_DISABLED
 #ifdef WITH_THREAD
```

Comment:
unrelated to this, but CPython removed threadless builds in Python 3.7, so `WITH_THREAD` is always defined